File: ancuta-gheorghe/RentingMovies/src/Movie.java
/**
 * Created with IntelliJ IDEA.
 * User: ancuta
 * Date: 5/18/13
 * Time: 5:36 PM
 * To change this template use File | Settings | File Templates.
 */
public class Movie {

    public static final int CHILDREN = 2;
    public static final int REGULAR = 0;
    public static final int NEW_RELEASE = 1;

    private String _title;
    private int _priceCode;

    public Movie(String title, int priceCode) {
        this._title = title;
        this._priceCode = priceCode;
    }

    public int getPriceCode() {
        return _priceCode;
    }

    public void setPriceCode(int arg) {
        this._priceCode = arg;
    }

    public String getTitle() {
        return _title;
    }

}

File: cristina-condruz/RefactoringExamples/src/Movie.php
<?php
/**
 * Created by JetBrains PhpStorm.
 * User: Cristina Condruz
 * Date: 5/28/13
 * Time: 10:53 AM
 * To change this template use File | Settings | File Templates.
 */
require_once('Price.php');

class Movie{
  public static $CHILDREN = 2;
  public static $REGULAR = 0;
  public static $NEW_RELEASE = 1;

  private $_title;
  private $_priceCode;
  private $_price; //Price type

  function __construct($_priceCode, $_title) {
    $this->setPriceCode($_priceCode);
    $this->_title = $_title;
  }

  public function setPriceCode($priceCode) {
    switch ($priceCode) {
       case Movie::$REGULAR:
         $this->_price = new RegularPrice();
         break;
       case  Movie::$NEW_RELEASE:
         $this->_price = new NewReleasePrice();
         break;
       case Movie::$CHILDREN:
         $this->_price = new ChildrenPrice();
         break;
      default:
        throw new Exception("Incorrect Price Code.");
     }
  }

  public function getPriceCode() {
    return $this->_price->getPriceCode();
  }

  public function setTitle($title) {
    $this->_title = $title;
  }

  public function getTitle() {
    return $this->_title;
  }

  /**
   * @param $daysRented
   * @return float|int
   */

  public function getCharge($daysRented){
    return $this->_price->getCharge($daysRented);
   }

  /**
   * @param $daysRented
   * @return int
   */
  public function getFrequentRenterPoints($daysRented) {
     return $this->_price->getFrequentRenterPoints($daysRented);
  }
}